luigi/util.py

```diff
@@ -313,6 +316,41 @@ def clone_parents(_self, **kwargs):
 
         return task_that_inherits
 
+    def check_for_parameter_collisions(self, task_that_inherits):
+        error_msg = (
+            "Parameter name collision detected in tasks_to_inherit. "
+            "Parameter '{param}' in '{task}' duplicates "
+            "parameter '{param}' in {another_task}. "
+            "Either rename one of the parameters or "
+            "include the '{param}' in the `ignore_collisions` list."
+        )
+
+        # Check that the parameters from the inheriting task don't mask any
+        # parameters from the inherited tasks.
+        for task_to_inherit in self.tasks_to_inherit:
+            for param_name, param_obj in task_to_inherit.get_params():
+                if (
+                    hasattr(task_that_inherits, param_name) and
+                    param_name not in self.ignore_collisions
+                ):
+                    raise AttributeError(error_msg.format(param=param_name,
+                                                          task=task_that_inherits,
+                                                          another_task=task_to_inherit))
+
+        # Check that the parameters from an inherited task don't mask the
+        # parameters from another inherited task.
+        for task_to_inherit in self.tasks_to_inherit:
```

Comment:
As written, this loop is ran 3 times. 2 within this method and once afterward. Is there anyway to combine the looping of these `tasks_to_inherit` to avoid repetitive looping?